inspirehep/modules/orcid/api.py

```diff
@@ -63,12 +70,19 @@ def push_record_with_orcid(recid, orcid, oauth_token, put_code=None, old_hash=No
     """
     record = get_db_record('lit', recid)
 
-    new_hash = calculate_hash_for_record(record)
+    orcid_cache = OrcidCache(orcid)
+
+    if not putcode or not old_hash:
+        cached_putcode, cached_old_hash = orcid_cache.read_record_data(recid)
+        putcode = putcode or cached_putcode
+        old_hash = old_hash or cached_old_hash
+
+    new_hash = OrcidHasher().compute_hash(record)
```

Comment:
Why not make it a class/static method?

Reply:
What's the benefit of that? I don't have a strong opinion here, but the fact that this can be a class method does not mean that it has to. And for regular use cases, I tend to avoid class/static methods such that the class has a very common signature.
By very common signature I mean:
```
ferrari = Car()
ferrari.start()
```

By non regular use cases, I mean f.i. this: https://github.com/inspirehep/inspire-next/pull/3506/commits/ac2963af9a641b5f6e843fd27c91ccb23aa75f77#r197426458
where I want to be able to do a `DuplicatedExternalIdentifiersError.match(myexc)` because I am interested in matching with the exception class and I do not want to create an instance.